model/Service/Result/SyncEncryptedResultService.php

```diff
@@ -109,6 +111,10 @@ public function importDeliveryResults(array $results)
                 $success = false;
             }
 
+            if ($success == true) {
```

Comment:
My suggestion it's to trigger Result Event after the decryption of result
1. `https://github.com/oat-sa/extension-tao-encryption/blob/481519beef1c8f00243f51145885798fe352578a/model/Service/Result/DecryptResultService.php#L119` 
Maybe here,

2. Remove the trigger of the event from here and move it to 
`https://github.com/oat-sa/extension-tao-sync/blob/8cd3e27644c6e775c3013b886dbe46f0c6d98ed0/model/ResultService.php#L237`
We want to support taoSync without taoEncryption at some point, moving it here it will be easy later